### DISCUSSION TITLE
encapsulating store within a hook

### DISCUSSION THREAD

**Discussion by jestrickler** [pmndrs/zustand#2767] [Q&A]
Would the following work?
```
const useStore = () =>
    const useZustandStore = create((set, get) => ({
        disabled: false,
        setDisabled: (disabled) => set(() => ({ disabled })),
        ...
    }))

    return {
        disabled: useZustandStore((state) => state.disabled),
        setDisabled: useZustandStore((state) => state.setDisabled),
        ....
    }
}

const Component = () => {
    const { disabled, setDisabled } = useStore()
    ...
}
```

**Comment by dbritto-dev** [comment]
@jestrickler yeah, that should work. Let us know if doesn't work.

**Comment by jestrickler** [comment]
It worked after I moved 'useZustandStore = create...' outside of the hook, so it's not creating a new store every time the hook is called, D'oh!! But I wonder if this is the same as fetching everything - const state = useZustandStore(), since the hook returns all the slices, and the component destructures the items it needs..?
